qiskit_optimization/problems/indicator_constraint.py

```diff
@@ -0,0 +1,181 @@
+        """Set the active value for the binary indicator variable of the constraint.
+
+        Returns:
+            The active value
+        """
+        self._active_value = active_value
+
+    @property
+    def binary_var(self) -> Variable:
+        """Returns the binary indicator variable of the constraint.
+
+        Returns:
+            The binary indicator variable
+        """
+        return self._binary_var
+
+    @binary_var.setter
+    def binary_var(self, binary_var: Variable) -> None:
+        """Set the binary indicator variable of the constraint.
+
+        Args:
+            binary_var: The binary indicator variable of the constraint.
+        """
+        self._binary_var = binary_var
+
+    @property
+    def linear(self) -> LinearExpression:
+        """Returns the linear expression corresponding to the left-hand-side of the constraint.
+
+        Returns:
+            The left-hand-side linear expression.
+        """
+        return self._linear
+
+    @linear.setter
+    def linear(
+        self,
+        linear: Union[ndarray, spmatrix, List[float], Dict[Union[str, int], float]],
+    ) -> None:
+        """Sets the linear expression corresponding to the left-hand-side of the constraint.
+        The coefficients can either be given by an array, a (sparse) 1d matrix, a list or a
+        dictionary.
+
+        Args:
+            linear: The linear coefficients of the left-hand-side.
+        """
+
+        self._linear = LinearExpression(self.quadratic_program, linear)
+
+    def evaluate(self, x: Union[ndarray, List, Dict[Union[int, str], float]]) -> float:
+        """Evaluate the left-hand-side of the constraint.
+
+        Args:
+            x: The values of the variables to be evaluated.
+
+        Returns:
+            The left-hand-side of the constraint given the variable values.
+        """
+        return self.linear.evaluate(x)
+
+    def evaluate_indicator(self, x: Union[ndarray, List, Dict[Union[int, str], float]]) -> bool:
+        """Evaluate the binary indicator var using the active value.
+
+        Args:
+            x: The values of the variables to be evaluated.
+
+        Returns:
+            If x equals to the active value, return True. Otherwise, return False.
+
+        Raises:
+            QiskitOptimizationError: if the given variable index does not match to the index of
+            self.binary_var.
+            QiskitOptimizationError: if x is given in unsupported format.
+        """
+        index = self.quadratic_program.variables_index[self.binary_var.name]
+        if isinstance(x, (list, ndarray)):
+            val = x[index]
+        elif isinstance(x, dict):
+            for ind, value in x.items():
+                if isinstance(ind, str):
+                    ind = self.quadratic_program.variables_index[ind]
+                if index == ind:
+                    val = value
+        else:
+            raise QiskitOptimizationError("Unsupported format for x.")
```

Comment:
Should the message, since its more related to type, not format, of x be more like 
`f"Unsupported type '{type(x)}' for x."` so it says type and we include the type in the message